internal/ceb/app_config.go

```diff
@@ -51,11 +54,28 @@ func (ceb *CEB) watchAppConfig(
 	// keeps track of all the dynamic configurations that we have.
 	var static []string
 	var dynamic map[string][]*component.ConfigRequest
+	var dynamicSources map[string]*pb.ConfigSource
 
 	// refreshCh will be sent a message when we want to refresh our
 	// configuration. We default to nil so that we do nothing until
 	// we receive our first set of variables (the <-inCh case below).
-	var refreshCh <-chan time.Time
+	//
+	// coalesceCh is used when we want to refresh, but allow some time
+	// for coalescing of the source/variable channels to occur.
+	var refreshCh, coalesceCh <-chan time.Time
+	refreshTick := func() {
+		// If we haven't scheduled a forced refresh, then schedule that.
+		// We will refresh NO MATTER WHAT on this timer and prevents a
+		// flurry of config updates from preventing variable refresh.
+		if refreshCh == nil {
+			refreshCh = time.After(5 * time.Second)
+		}
+
+		// Reset our coalesce channel. Using "time.After" here "leaks"
+		// timers if we're calling this enough but they're a bunch of timers
+		// that reset relatively quickly so let's just let it happen for now.
+		coalesceCh = time.After(500 * time.Millisecond)
```

Comment:
Per the note about leaking timer, could we use a persistent `Ticker` and call `Ticker.Reset` here to boost it to 500ms?

Reply:
If you do use a persistent `Ticker` here, I'd still assign it's `.C` to `coalesceCh` so that you can set it to `nil` later on.

Reply:
I avoided Tickers and Timers because I found their reset/stop logic fairly easy to mess up (the docs have all sorts of caveats on when they should be called and I've messed it up many times). I prefer to use leaky "After" unless we think it's a frequent event. In this case, I think changes to config should be infrequent enough that I'd prefer to use this since I know it works...